pkg/filter/component.go

```diff
@@ -0,0 +1,58 @@
+package filter
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/asecurityteam/awsconfig-filterd/pkg/domain"
+)
+
+// FilterConfig contains settings for the FilterComponent.
+type FilterConfig struct {
+	Enabled      []string `description:"List of enabled filters."`
+	ResourceType *ResourceTypeConfig
+}
+
+// Name of the configuration root.
+func (*FilterConfig) Name() string {
+	return "filter"
+}
+
+// FilterComponent is the top-level container for all filter types.
+type FilterComponent struct {
+	ResourceType *ResourceTypeComponent
+}
+
+// NewFilterComponent generates a FilterComponent.
+func NewFilterComponent() *FilterComponent {
+	return &FilterComponent{
+		ResourceType: NewResourceTypeComponent(),
+	}
+}
+
+// Settings generates the default configuration.
+func (c *FilterComponent) Settings() *FilterConfig {
+	return &FilterConfig{
+		Enabled:      []string{"resourcetype"},
+		ResourceType: c.ResourceType.Settings(),
+	}
+}
+
+// New generates a filter.
+func (c *FilterComponent) New(ctx context.Context, conf *FilterConfig) (domain.ConfigFilterer, error) {
+	f := make(AnyMatch, 0, len(conf.Enabled))
```

Comment:
nit: Because I missed this on my first reading of this, might be worth tossing a comment here to draw extra attention to the use of `AnyMatch` instead of the vanilla `[]domain.ConfigFilterer`.